frontend/src/Admin/components/AssignJudgesTable.js

```diff
@@ -0,0 +1,138 @@
+import React, { Component } from 'react'
+import { Row, Col, Button } from 'reactstrap'
+import { Flex, Box } from 'rebass'
+import styled from 'styled-components'
+import PropTypes from 'prop-types'
+import FaLongArrowLeft from 'react-icons/lib/fa/long-arrow-left'
+import FaLongArrowRight from 'react-icons/lib/fa/long-arrow-right'
+
+import JudgesTable from '../components/JudgesTable'
+
+const CenteredSubHeading = styled.h2`
+  text-align: center;
+`
+
+class AssignJudgesTable extends Component {
+  static propTypes = {
+    showId: PropTypes.string.isRequired,
+    fetchData: PropTypes.func.isRequired,
+    data: PropTypes.shape({
+      unassignedJudges: PropTypes.array.isRequired,
+      assignedJudges: PropTypes.array.isRequired
+    }),
+    assign: PropTypes.func.isRequired,
+    unassign: PropTypes.func.isRequired,
+    afterAssign: PropTypes.func.isRequired,
+    afterUnassign: PropTypes.func.isRequired
+  }
+
+  static defaultProps = {
+    data: {
+      assignedJudges: [],
+      unassignedJudges: []
+    }
+  }
+
+  state = {
+    selectedUnassignedJudges: {},
+    selectedAssignedJudges: {}
+  }
+
+  componentDidMount () {
+    this.props.fetchData()
+  }
+
+  assign = () => {
+    const judges = Object.keys(this.state.selectedUnassignedJudges)
+
+    if (judges.length) {
+      this.props.assign(judges)
+        .then(() => {
+          this.props.afterAssign(judges)
+          // Reset the checkboxes
+          this.setState({
+            selectedUnassignedJudges: {},
```

Comment:
Why do you ask?

The reason I do this is because data for both tables is refetched after assigning/unassigning Judges, so to keep both tables selection state in sync, I clear them after an action. The alternative is to do some weird merging / removing of incoming and outgoing data in each tables which is not worth the effort. Clearing the selections feels to have decently good UX.